include/zephyr/spinlock.h

```diff
@@ -44,8 +44,24 @@ struct z_spinlock_key {
  */
 struct k_spinlock {
```

Comment:
All the fields in this API struct need to have proper Doxygen comments or be marked as internal if they are internal. See e.g. include/zephyr/kernel/stats.h on how to document fields guarded by ifdefs so that they show up in the documentation while also indicating in which configuration they may be available.

Reply:
Hi @kartben ,

I think the fields of this struct are indeed internal, the user shouldn't change these fields directly. 
But at the same time struct k_spinlock should be mentioned in documentation.. 

So we just need to mark the internals of this struct with 

```
struct k_spinlock {
/**
 * @cond INTERNAL_HIDDEN
 */
....
/**
 * INTERNAL_HIDDEN @endcond
 */
};
```

Is it right?

Reply:
Hi @kartben , could you please comment?

Reply:
> So we just need to mark the internals of this struct with
> 
> ```
> struct k_spinlock {
> /**
>  * @cond INTERNAL_HIDDEN
>  */
> ....
> /**
>  * INTERNAL_HIDDEN @endcond
>  */
> };
> ```
> 
> Is it right?

absolutely!